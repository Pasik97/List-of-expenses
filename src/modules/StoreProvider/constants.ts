import { ReactNode } from "react";
import Stores from "stores/Stores";

export interface StoreProviderProps {
   store: Stores;
   children: ReactNode;
}
