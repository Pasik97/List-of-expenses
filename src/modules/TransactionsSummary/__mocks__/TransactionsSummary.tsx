const mock = () => '(TransactionsSummary /)';

export default mock;
