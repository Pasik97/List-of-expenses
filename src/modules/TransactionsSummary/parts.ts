import styled from 'styled-components';

export const Summary = styled.p`
   color: #1a1a1a;
   font-size: 18px; 
   line-height: 21px;
   margin: 16px 0 0 0;
`;
