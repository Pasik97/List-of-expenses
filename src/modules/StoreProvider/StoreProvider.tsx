import React, { createContext, } from 'react';
import { StoreProviderProps } from './constants';
import Stores from 'stores/Stores';

export const StoreProviderContext = createContext<Stores>({} as Stores);

const StoreProvider: React.FC<StoreProviderProps> = ({
   children,
   store,
}) => (
   <StoreProviderContext.Provider value={store}>
      {children}
   </StoreProviderContext.Provider>
);

export default StoreProvider;
