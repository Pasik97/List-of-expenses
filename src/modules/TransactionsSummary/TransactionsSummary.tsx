import React, { useContext } from 'react';
import { observer } from 'mobx-react';
import * as P from './parts';
import { StoreProviderContext } from 'modules/StoreProvider/StoreProvider';

const TransactionsSummary: React.FC = observer(() => {
   const { transactions } = useContext(StoreProviderContext);
   const { pln, fixedEur } = transactions.transactionsSummay;

   return !pln ? null : (
      <P.Summary>Sum: {pln} PLN ({fixedEur} EUR)</P.Summary>
   )
});

export default TransactionsSummary;
