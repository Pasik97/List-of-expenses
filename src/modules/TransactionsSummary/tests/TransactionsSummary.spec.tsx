import { create } from "react-test-renderer";
import * as M from './mockData';
import TransactionsSummary from "../TransactionsSummary";
import StoreProvider from "modules/StoreProvider/StoreProvider";

describe('<TransactionsSummary />', () => {
   it('should render summary of pln and eur amount', () => {
      const component = create(
         <StoreProvider store={M.mockStore}>
            <TransactionsSummary />
         </StoreProvider>,
      )

      expect(component.toJSON()).toMatchSnapshot();
   });

   it('should render null if pln amount is 0', () => {
      const component = create(
         <StoreProvider store={M.mockStoreWithEmptyTransactionSummary}>
            <TransactionsSummary />
         </StoreProvider>,
      )

      expect(component.toJSON()).toBeNull();
   });
});
