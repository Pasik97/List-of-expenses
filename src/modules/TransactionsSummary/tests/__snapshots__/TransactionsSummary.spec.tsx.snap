// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransactionsSummary /> should render summary of pln and eur amount 1`] = `
<p
  className="sc-bdvvtL dLAFml"
>
  Sum: 
  10
   PLN (
  20
   EUR)
</p>
`;
